### ISSUE TITLE
Other install options

### ISSUE THREAD

**Issue by SylvanFranklin** [Thumuss/utpm#33]
It would be great to have this on crates.io so that users could simply install using `cargo install utpm`. I don't know if you have any interest in other package managers, but I am on a mac and have experience with homebrew and could create a tap for `brew install`.

**Comment by corrosive4354** [comment]
Seeing that you are using homebrew, I would like to recommend you the tool [Servbay](servbay.com), which is more professional and easier to use on Mac than Docker and homebrew.

**Comment by SylvanFranklin** [comment]
Clearly a promotion attempt. That is not a drop in replacement to either docker or home brew both of which are much more established, and have different use cases. Also that link doesn't work and the website and repo is full of spelling mistakes.

**Comment by Thumuss** [comment]
Good idea! 
I already had an interest in it but since they are a LOT of package manager I don't really know where to start but it's definitely in my todo list!

**Comment by SylvanFranklin** [comment]
Yes you are right there are so many different platforms to consider. I think cargo is the most important since Typst developers almost certainly will have Rust installed. You already have a way to install it with cargo currently which is great, I think `cargo binstall` or a regular `cargo install` would still be good in the future. I'll see about getting a homebrew tap working in the next few days but I don't see it as a huge priority.

**Comment by Thumuss** [comment]
A friend of mine will (potentially) add nix support here!

**Comment by SylvanFranklin** [comment]
Any progress on an upload to crates.io?

**Comment by Thumuss** [comment]
It's a bit complicated. 
Tinger had made a crate named `typst-project` that I've used for a long time but it is not published directly to crates.io.
They made two new crates named `typst-syntax` and `typst-kit` which contains a bit of `typst-project` but not all of it.
So I need to rewrite pretty much 50% of the code that I have to publish it.

As my finals are coming, I think I'll do it a bit later.

Meanwhile I'll finish publish command this month or in january.
I'll check debian, ubuntu, arch and fedora packaging system to add `utpm` on them!

**Comment by SillyFreak** [comment]
I think mirroring what [tytanic](https://github.com/typst-community/tytanic) does would be valuable:

- release to crates.io (once the blockers are resolved)
- add Github releases with binaries for various platforms, via a [Github action](https://github.com/typst-community/tytanic/blob/main/.github/workflows/release.yml))

then you can install using `cargo binstall`, which is much quicker (and energy saving) than `cargo install`. And it would enable using utpm in the template's CI (next step: `setup-utpm` action).

**Comment by Thumuss** [comment]
On the last PR #83 I've added a lot of packages manager. If you want to look at that, feel free!

**Comment by frabera** [comment]
> On the last PR [#83](https://github.com/typst-community/utpm/pull/83) I've added a lot of packages manager. If you want to look at that, feel free!

I added the package to scoop for installation in windows: [https://scoop.sh/#/apps?q=utpm](https://scoop.sh/#/apps?q=utpm) [https://github.com/ScoopInstaller/Extras/blob/master/bucket/utpm.json](https://github.com/ScoopInstaller/Extras/blob/master/bucket/utpm.json)

Actually it takes the releases from github so it's at v0.2.0 and currently it panics on windows.

**Comment by Thumuss** [comment]
I _should_ have time since I don't have a lot going now right now. This weekend I'll try to publish 0.3
